core/src/main/scala/kafka/api/TopicMetadata.scala

```diff
@@ -60,26 +61,26 @@ case class TopicMetadata(topic: String, partitionsMetadata: Seq[PartitionMetadat
   override def toString(): String = {
     val topicMetadataInfo = new StringBuilder
     topicMetadataInfo.append("{TopicMetadata for topic %s -> ".format(topic))
-    errorCode match {
-      case ErrorMapping.NoError =>
+    Errors.forCode(errorCode) match {
+      case Errors.NONE =>
         partitionsMetadata.foreach { partitionMetadata =>
-          partitionMetadata.errorCode match {
-            case ErrorMapping.NoError =>
+          Errors.forCode(partitionMetadata.errorCode) match {
+            case Errors.NONE =>
               topicMetadataInfo.append("\nMetadata for partition [%s,%d] is %s".format(topic,
                 partitionMetadata.partitionId, partitionMetadata.toString()))
-            case ErrorMapping.ReplicaNotAvailableCode =>
+            case Errors.REPLICA_NOT_AVAILABLE =>
               // this error message means some replica other than the leader is not available. The consumer
               // doesn't care about non leader replicas, so ignore this
               topicMetadataInfo.append("\nMetadata for partition [%s,%d] is %s".format(topic,
                 partitionMetadata.partitionId, partitionMetadata.toString()))
             case _ =>
               topicMetadataInfo.append("\nMetadata for partition [%s,%d] is not available due to %s".format(topic,
-                partitionMetadata.partitionId, ErrorMapping.exceptionFor(partitionMetadata.errorCode).getClass.getName))
+                partitionMetadata.partitionId, Errors.forCode(partitionMetadata.errorCode).exception.getClass.getName))
           }
         }
       case _ =>
         topicMetadataInfo.append("\nNo partition metadata for topic %s due to %s".format(topic,
-                                 ErrorMapping.exceptionFor(errorCode).getClass.getName))
+          Errors.forCode(errorCode).exception.getClass.getName))
```

Comment:
In the code above, we are calling `Errors.forCode` more times than necessary as we are discarding the pattern match variable (two instances that I can see).